@xen-orchestra/web-core/lib/components/ui/table-pagination/UiTablePagination.vue

```diff
@@ -0,0 +1,212 @@
+<!-- v2 -->
+<template>
+  <div class="ui-table-pagination">
+    <span class="typo p3-regular label">
+      {{ $t('core.select.n-object-of', { from: startIndex, to: endIndex, total: totalItems }) }}
+    </span>
+
+    <div class="buttons-content">
+      <UiButtonIcon
+        :disabled="isFirstPage"
+        accent="info"
+        class="icon"
+        size="small"
+        :icon="faAngleDoubleLeft"
+        @click="goToFirstPage"
+      />
+      <UiButtonIcon
+        :disabled="isFirstPage"
+        accent="info"
+        class="icon"
+        size="small"
+        :icon="faAngleLeft"
+        @click="goToPreviousPage"
+      />
+      <UiButtonIcon
+        :disabled="isLastPage"
+        accent="info"
+        class="icon"
+        size="small"
+        :icon="faAngleRight"
+        @click="goToNextPage"
+      />
+      <UiButtonIcon
+        :disabled="isLastPage"
+        accent="info"
+        class="icon"
+        size="small"
+        :icon="faAngleDoubleRight"
+        @click="goToLastPage"
+      />
+    </div>
+
+    <span class="typo p3-regular label">{{ $t('core.show-by') }}</span>
+    <div class="dropdown-wrapper">
+      <select v-model="itemsPerPage" :disabled class="dropdown typo c3-regular" @change="updatePagination">
+        <option v-for="option in pageSizeOptions" :key="option" :value="option" class="typo p2-medium">
+          {{ option }}
+        </option>
+      </select>
+      <VtsIcon class="icon" accent="current" :icon="faAngleDown" />
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import VtsIcon from '@core/components/icon/VtsIcon.vue'
+import UiButtonIcon from '@core/components/ui/button-icon/UiButtonIcon.vue'
+import {
+  faAngleDoubleLeft,
+  faAngleDoubleRight,
+  faAngleDown,
+  faAngleLeft,
+  faAngleRight,
+} from '@fortawesome/free-solid-svg-icons'
+import { ref, computed } from 'vue'
+
+const { totalItems } = defineProps<{
+  totalItems: number
+  disabled?: boolean
+}>()
+
+const currentPage = ref(1)
+const itemsPerPage = ref(50)
+const pageSizeOptions = [10, 50, 100, 150, 200]
+
+const totalPages = computed(() => Math.ceil(totalItems / itemsPerPage.value))
+const startIndex = computed(() => (currentPage.value - 1) * itemsPerPage.value + 1)
+const endIndex = computed(() => Math.min(currentPage.value * itemsPerPage.value, totalItems))
+
+const isFirstPage = computed(() => currentPage.value === 1)
+const isLastPage = computed(() => currentPage.value === totalPages.value)
+
+const goToFirstPage = () => {
+  currentPage.value = 1
+}
+
+const goToPreviousPage = () => {
+  if (currentPage.value > 1) currentPage.value -= 1
+}
+
+const goToNextPage = () => {
+  if (currentPage.value < totalPages.value) currentPage.value += 1
+}
+
+const goToLastPage = () => {
+  currentPage.value = totalPages.value
+}
```

Comment:
Even if it is simple, maybe we can use [this utility](https://vueuse.org/core/useOffsetPagination/#useoffsetpagination) from VueUse which would provide more functionality.